### PR TITLE
Add init command to prepare a new project for mitre

### PR DESCRIPTION
**Pull request by DanielMSchmidt**

### CODE REVIEW

File: src/bin/mitre.rs [hunk 1]
```diff
@@ -61,6 +62,75 @@ fn main() {
     // Validate config contains a mitre runner
 
     match m.subcommand_name() {
+        Some("init") => {
+            let config_path = m
+                .value_of("config_file")
+                .unwrap_or(mitre::DEFAULT_CONFIG_FILE);
+            let migrations_dir = m.value_of("directory").unwrap_or("./migrations");
```
Comment (leehambley) [inline src/bin/mitre.rs:69]: You shouldn't need to do this, those are set a bit higher up (around #50-55) and that variable should be in scope here.

File: src/bin/mitre.rs [hunk 2]
```diff
@@ -61,6 +62,75 @@ fn main() {
...
+                let migrations_readme = format!(
+                    "# Mitre Migrations
+This directory contains migrations to be used with [Mitre](https://github.com/leehambley/mitre).
+## Getting Started
+To run the migrations in this folder run
+```sh
+# Getting the current state
+mitre -c {} -d {} ls
+# See all commands
+mitre --help
```
Comment (leehambley) [inline src/bin/mitre.rs:109]: Nice explainer :taco:

File: src/config.rs [hunk 3]
```diff
@@ -283,6 +283,39 @@ pub fn from_file(p: &Path) -> Result<Configuration, ConfigError> {
...
+  username: root
+
+# Curl Runner Data Stores can not be used as mitre config
+my-elasticsearch:
+  _runner: curl
+  ip_or_hostname: es
+  protocol: http
+  logLevel: debug
+
+# The key mitre signals that this data store is going to be used for mitres migration table
```
Comment (leehambley) [inline src/config.rs:310]: FYI I'm struggling a bit here, after the latest WIP about bootstrapping, the MariaDB runner is looking for a "mariadb" config on here, and I *think* I need to make it look for a `config.get("mitre")` (not `"mariadb"`) and then check that the `config._runner` is equal to `mariadb`. Didn't want to decide anything without you.

File: src/config.rs [hunk 4]
```diff
@@ -502,4 +536,27 @@ a:
             }
         }
     }
+
+    #[test]
+    fn default_config_is_valid() -> Result<(), String> {
```
Comment (leehambley) [inline src/config.rs:541]: Nice.

**Review by leehambley — APPROVED** [review]
Looking really good, happy to merge this now.

btw we can also add tests in `bin/mitre.rs` for some of the CLI/CLAP stuff I guess.

**Comment by DanielMSchmidt** [comment]
Yeah, lets add some CLI tests somewhen soon!

**Comment by leehambley** [comment]
I looked into https://github.com/sstephenson/bats too, seemed OK, but I think we can do BATS integration tests, and also some tests in the `mitre.rs` executable tests block too.

**Pull request merged**